File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> um;
        for (int i = 0; i < nums.size(); i++) {
            int x = target - nums[i];
            if (um.find(x) != um.end()) return {um[x], i};
            um[nums[i]] = i;
        }

        return {0, 0};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        track = {}

        for idx, n in enumerate(nums):
            comp = target - n

            if comp in track:
                return [track[comp], idx]
            
            track[n] = idx

        return []